classes/security/authorization/AssignedStageRoleHandlerOperationPolicy.php

```diff
@@ -59,13 +63,49 @@ public function effect()
     {
         // Check whether the user has one of the allowed roles
         // assigned. If that's the case we'll permit access.
-        // Get user roles grouped by context.
-        $userRoles = $this->getAuthorizedContextObject(Application::ASSOC_TYPE_ACCESSIBLE_WORKFLOW_STAGES);
```

Comment:
As far as I understand, we don't need workflow stages, which are accessible for the user here, as they also include stages accessible because of local roles, like admin or manager. Instead stage assignments should be used.